zipkin-lens/src/components/App/App.test.js

```diff
@@ -11,54 +11,61 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-import { shallow } from 'enzyme';
 import React from 'react';
 import { Provider } from 'react-redux';
 import { BrowserRouter, Route } from 'react-router-dom';
 import { ThemeProvider } from '@material-ui/styles';
 import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { createMount, createShallow } from '@material-ui/core/test-utils';
 
 import App from './App';
 import Layout from './Layout';
+import DiscoverPage from '../DiscoverPage';
+import TracePage from '../TracePage';
+import { theme } from '../../colors';
 
 describe('<App />', () => {
-  let wrapper;
+  let mount;
+  let shallow;
 
   beforeEach(() => {
-    wrapper = shallow(<App />);
+    mount = createMount({ strict: true });
+    shallow = createShallow();
   });
 
-  it('should render Layout', () => {
-    const items = wrapper.find(Layout);
-    expect(items.length).toBe(1);
+  afterEach(() => {
+    mount.cleanUp();
   });
 
-  describe('should render router', () => {
-    it('should render BrowserRouter', () => {
-      const items = wrapper.find(BrowserRouter);
-      expect(items.length).toBe(1);
-    });
+  it('should update document title', () => {
+    mount(<App />);
+    expect(document.title).toBe('Zipkin');
+  });
 
-    it('should render 2 routes', () => {
-      const items = wrapper.find(Route);
-      expect(items.length).toBe(2);
-    });
+  it('should render providers', () => {
```

Comment:
this test says what not why. providers are a generic term. Ex. 'should provide only one theme' or something.

then, below, have a comment for each and what they control. This will give the why about this test. In other words, I think there are separate goals.. coverage is one and why we code is another. The latter is more thought even if the former more work.